### ISSUE TITLE
Model Search form: Inaccessible close button

### ISSUE THREAD

**Issue by kangabell** [danny-avila/LibreChat#5564]
The close button in the search form for LLM Models is not accessible. It is not reachable by keyboard or screen reader and is not described or labeled as a button programmatically. 	

## Testing Environment
http://159.89.83.1/c/new 	

## WCAG Criteria
2.4.6 Headings and Labels - AA	

## Testing Method
Manual - Chrome/MacOS	

## Screenshot

![A button with the letter "x" is highlighted, and an arrow points to the associated element in the browser inspector.](https://github.com/user-attachments/assets/68e21558-0a68-49fb-90a5-6be22a3997ea)

## Code Snippet
```<div class="relative flex h-5 w-5 items-center justify-end text-gray-500"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-x text-gray-500 dark:text-gray-300 cursor-pointer opacity-100"><path d="M18 6 6 18"></path><path d="m6 6 12 12"></path></svg></div>```

## Recommended Action
The close button in the search bar for the LLM Models is not a programmatic button and is not labeled with an accessible name. Use the <button> element to make this a semantic button that screen readers and keyboard users can access. Also provide a name like "close LLM Model search" to describe it. 	

## Notes
This is a barrier identifed in Fable testing where a user was unable to interact with, or use the search feature successfully.
